Content.Server/DeltaV/ProxyDetection/DetectionManager.cs

```diff
@@ -0,0 +1,128 @@
+            { "ip", addr.ToString() },
+            { "vpn-lookup", "true" }
+        };
+        var probeParameters = new Dictionary<string, string>
+        {
+            { "ip", addr.ToString()}
+        };
+
+        var blacklistResponse = await _neutrinoApiClient.IpBlocklist(blacklistParameters);
+        if (!blacklistResponse.IsOk())
+        {
+            _sawmill.Error("API Error: {0}, Error Code: {1}, HTTP Status Code: {2}",
+                blacklistResponse.ErrorMessage,
+                blacklistResponse.ErrorCode.ToString(),
+                blacklistResponse.HttpStatusCode.ToString()); // you should handle this gracefully!
+            _sawmill.Error($"{blacklistResponse.ErrorCause}");
+            return (false, "");
+        }
+
+        var data = blacklistResponse.Data;
+        data.TryGetProperty("blocklists", out var blockList);
+        data.TryGetProperty("is-listed", out var isListed);
+
+        if (!isListed.GetBoolean() && !_shouldProbe)
+            return (false, "");
+
+        string result;
+
+        if (!isListed.GetBoolean() && _shouldProbe)
+        {
+            var probeResponse = await _neutrinoApiClient.IpProbe(probeParameters);
+            if (!probeResponse.IsOk())
+            {
+                _sawmill.Error("API Error: {0}, Error Code: {1}, HTTP Status Code: {2}",
+                    probeResponse.ErrorMessage,
+                    probeResponse.ErrorCode.ToString(),
+                    probeResponse.HttpStatusCode.ToString()); // you should handle this gracefully!
+                _sawmill.Error($"{probeResponse.ErrorCause}");
+                return (false, "");
+            }
+
+            var probeData = probeResponse.Data;
+            probeData.TryGetProperty("is-proxy", out var isProxy);
+            probeData.TryGetProperty("is-hosting", out var isHosting);
+            probeData.TryGetProperty("is-vpn", out var isVpn);
+            if (!isProxy.GetBoolean() && !isHosting.GetBoolean() && !isVpn.GetBoolean())
+                return (false, "");
+
+            result = "Suspicious connection.";
+        }
+        else
+        {
+            var blockListsArray = blockList.EnumerateArray().Select(item => item.GetString()).ToList();
+            var blockLists = string.Join(", ", blockListsArray);
+
+            result = $"Your address was found in the following blacklists: {blockLists}";
+        }
+
+        var hid = addr.AddressFamily == AddressFamily.InterNetworkV6 ? 128 : 32;
+        _banManager.CreateServerBan(e.UserId, e.UserName, null, (addr, hid), null, null, NoteSeverity.High, result,
+            ServerBanExemptFlags.Datacenter);
+        return (true, result);
+    }
+
+    void IPostInjectInit.PostInject()
+    {
+        _sawmill = _log.GetSawmill("PROXYDETECTION");
+
+        if (_cfg.GetCVar(DCCVars.BlockProxyConnections))
+        {
+            if (_cfg.GetCVar(DCCVars.ProxyApiKey) == "" || _cfg.GetCVar(DCCVars.ProxyApiUser) == "")
+                return;
```

Comment:
This exactly probably won't work, but these ifs can be compacted to one
```suggestion
        if (_cfg.GetCVar(DCCVars.BlockProxyConnections) &&
            !(_cfg.GetCVar(DCCVars.ProxyApiKey) == "" || _cfg.GetCVar(DCCVars.ProxyApiUser) == ""))
```